src/plugins/intel_gpu/src/kernel_selector/cl_kernels/softmax_gpu_bf.cl

```diff
@@ -61,7 +61,7 @@ KERNEL (softmax_gpu_continuous_bfyx)(
     __local INPUT0_TYPE lg_storage[SLM_SIZE];
 
     uint i=0;
-    if (workers_per_data_set > SUB_GROUP_SIZE)
+    if (workers_per_data_set >= SUB_GROUP_SIZE && (leftovers % (16 / sizeof(INPUT0_TYPE))) == 0)
```

Comment:
1) Having additional calculation in cl kernel will not be good for performance. 
 We'd better have deterministic 
2) I don't see any reason to have a new PR from my PR. Would like to have a review comment to my PR, if you have any opinion,.. if the idea is completely different.

Reply:
Most of these calculations should be optimized out by compiler. In static shape mode the whole condition would be optimized out since all variables are either constants or macros. Stable diffusion Softmax doesn't demonstrate any degradation.